src/mail-app/mail/view/MailRow.ts

```diff
@@ -64,26 +65,34 @@ export class MailRow implements VirtualRow<Mail> {
 	private checkboxDom!: HTMLInputElement
 	private checkboxWasVisible = shouldAlwaysShowMultiselectCheckbox()
 	private selectionSetter!: SelectableRowSelectedSetter
+	private highlightedStrings: readonly SearchToken[] | null = null
 
 	constructor(
 		private readonly showFolderIcon: boolean,
 		private readonly getLabelsForMail: (mail: Mail) => ReadonlyArray<MailFolder>,
 		private readonly onSelected: (mail: Mail, selected: boolean) => unknown,
+		private readonly getHighlightedStrings?: () => readonly SearchToken[],
 	) {
 		this.top = 0
 		this.entity = null
 	}
 
 	update(mail: Mail, selected: boolean, isInMultiSelect: boolean): void {
+		const oldEntity = this.entity
 		this.entity = mail
+		const oldHighlightedStrings = this.highlightedStrings
+		this.highlightedStrings = this.getHighlightedStrings?.() ?? null
 
 		this.selectionSetter(selected, isInMultiSelect)
 		this.checkboxDom.checked = isInMultiSelect && selected
-
 		this.iconsDom.textContent = this.iconsText(mail)
 		this.dateDom.textContent = formatTimeOrDateOrYesterday(mail.receivedDate)
-		this.senderDom.textContent = getSenderOrRecipientHeading(mail, true)
-		this.subjectDom.textContent = mail.subject || NBSP
+
+		if (oldEntity !== this.entity || oldHighlightedStrings !== this.highlightedStrings) {
```

Comment:
did you test if it works in practice? I think we tend to rebind the rows very often e.g. during scrolling, even when there's no need to (which is also an issue worth addressing but anyway)